File: leetcode/day-20_pascals_triangle.py
from typing import List

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        triangle = [[1]]
        for i in range(numRows-1): #3: 0,1,2
            next_row = [1,]
            for j in range(1, len(triangle[i])): #3: 1,2 
                next_row.append(triangle[i][j-1] + triangle[i][j]) #3: next_row = trig[0][0] + trig[0][1]
            next_row.append(1) # next_row = [1,1+]
            triangle.append(next_row)
        return triangle
    

sol = Solution()

print(sol.generate(5))

File: leetcode/day-20_pascals_triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> triangle;
        if (numRows == 0)
            return triangle;
        triangle.push_back({1});
        
        for (int i = 1; i < numRows; ++i) {
            vector<int> nextRow = {1};
            for (int j = 1; j < triangle[i-1].size(); ++j) {
                nextRow.push_back(triangle[i-1][j-1] + triangle[i-1][j]);
            }
            nextRow.push_back(1);
            triangle.push_back(nextRow);
        }
        return triangle;
    }
};